modules/extension/xsd/xsd-core/src/main/java/org/geotools/xml/Encoder.java

```diff
@@ -743,8 +787,15 @@ public void encode(Object object, QName name, ContentHandler handler)
                             }
                         }
                         else {
-                            //add the next object to be encoded to the stack
-                            encoded.push(new EncodingEntry(next, element,entry));                            
+                            if (relaxed && isNonStrippedNestedElement(next, element)) {
```

Comment:
I think putting an instanceof test first would guard against performance degradation for non-complex cases (consistent with previous ```if``` clause) and is also great documentation. Yes, I saw that you also had a test in the method body, which is fine. Call me paranoid but this method is a real hot spot and avoiding method calls inside it might be important:
```
if (next instanceof ComplexAttribute && relaxed && isNonStripedNestedElement(next, element)) {
```
Note also the change ```isNonStrippedNestedElement``` -> ```isNonStripedNestedElement``` as mentioned above.

You can also fix the format of the previous two lines to be "} else {".  :-)